datafusion/core/src/datasource/file_format/write/mod.rs

```diff
@@ -149,15 +145,14 @@ impl<W: AsyncWrite + Unpin + Send> AsyncWrite for AbortableWrite<W> {
 
 /// A trait that defines the methods required for a RecordBatch serializer.
 #[async_trait]
-pub trait BatchSerializer: Unpin + Send {
+pub trait SerializationSchema: Sync + Send {
     /// Asynchronously serializes a `RecordBatch` and returns the serialized bytes.
-    async fn serialize(&mut self, batch: RecordBatch) -> Result<Bytes>;
-    /// Duplicates self to support serializing multiple batches in parallel on multiple cores
-    fn duplicate(&mut self) -> Result<Box<dyn BatchSerializer>> {
-        Err(DataFusionError::NotImplemented(
-            "Parallel serialization is not implemented for this file type".into(),
-        ))
-    }
+    async fn serialize(&self, batch: RecordBatch) -> Result<Bytes>;
```

Comment:
Instead of `duplicate_headerless` another pattern could be to pass in another parameter here, which matches how the calculation is done in the serialization code

```suggestion
   // is_initial is true if this is the first batch to be serialized by this serializer
    async fn serialize(&self, 
      batch: RecordBatch,
      is_initial: bool,
      ) -> Result<Bytes>;
```

Reply:
I think your suggestion is a better pattern and I will send a commit to incorporate it shortly